poollet/machinepoollet/controllers/machine_controller.go

```diff
@@ -93,37 +86,26 @@ func (r *MachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		if !apierrors.IsNotFound(err) {
 			return ctrl.Result{}, fmt.Errorf("error getting machine %s: %w", req.NamespacedName, err)
 		}
-		return r.deleteGone(ctx, log, req.NamespacedName)
+		return r.deleteGone(ctx, log, machine)
 	}
 	return r.reconcileExists(ctx, log, machine)
 }
 
 func (r *MachineReconciler) getIRIMachinesForMachine(ctx context.Context, machine *computev1alpha1.Machine) ([]*iri.Machine, error) {
-	labelKey := poolletutils.DownwardAPILabel(v1alpha1.MachineDownwardAPIPrefix, v1alpha1.RootMachineUIDLabelSuffix)
-	machineUID := machine.Labels[labelKey]
-	if machineUID == "" {
-		machineUID = string(machine.GetUID())
-	}
-	labels := map[string]string{
-		labelKey: machineUID,
-	}
-	res, err := r.MachineRuntime.ListMachines(ctx, &iri.ListMachinesRequest{
-		Filter: &iri.MachineFilter{LabelSelector: labels},
-	})
-	if err != nil {
-		return nil, fmt.Errorf("error listing machines by machine uid: %w", err)
-	}
-	return res.Machines, nil
-}
-
-func (r *MachineReconciler) listMachinesByMachineKey(ctx context.Context, machineKey client.ObjectKey) ([]*iri.Machine, error) {
-	res, err := r.MachineRuntime.ListMachines(ctx, &iri.ListMachinesRequest{
-		Filter: &iri.MachineFilter{LabelSelector: r.machineKeyLabelSelector(machineKey)},
-	})
-	if err != nil {
-		return nil, fmt.Errorf("error listing machines by machine key: %w", err)
+	if machine.Status.MachineID != "" {
```

Comment:
This check is kind of bad since in case of a controller crash we lose the connection between two `Machine`s.  We can not get over time in a sable state again. The scenario is not that unlikely, let's assume you create a new `Machine`
- `getIRIMachinesForMachine` has len 0 
- `func (r *MachineReconciler) create(` will triggered 
- `Machine` in the below cluster will be created
-  But something within fails: 
```
	log.V(1).Info("Updating status")
	if err := r.updateStatus(ctx, log, machine, res.Machine, nics); err != nil {
		return ctrl.Result{}, fmt.Errorf("error updating machine status: %w", err)
	}
```
In the next reconcile we again recreate the `Machine` which leads to a unstable state